### PR TITLE
Blackify osmo-jupyter

### PR DESCRIPTION
**Pull request by jason-curtis**
[asana ticket](https://app.asana.com/0/819671808102776/1127643074336798/f)

I did the following:

* checkout latest master
* create f/black-formatter branch
* run "black ."
* Review diff, use commented fmt: blocks and roll back anything egregious (search for `fmt` for how this looks. unfortunately I couldn't figure out a way to turn formatting off and comment it on one line)
* edit test_requirements.txt to add black, pytest-black
* edit setup.cfg to add flake8 ignores and --black pytest flag,
* edit README.md to add "code style: black" flag
* check that `pytest` is happy and that if I mess up formatting, `pytest` now displays the diff that `black` desires after all the test output
* commit
* open PR

## TODO:

same thing for cosmobot-process-experiment and cosmobot-run-experiment if this passes review
update code style manifesto if this passes review

### CODE REVIEW

File: osmo_jupyter/calibration/do/prep.py [hunk 1]
```diff
@@ -31,48 +28,47 @@ def prep_calibration_data(
...
     # Import camera data
-    all_camera_data = pd.read_csv(
-        image_data_filepath,
-        parse_dates=['timestamp']
-    )
+    all_camera_data = pd.read_csv(image_data_filepath, parse_dates=["timestamp"])
 
     # Pull out some choice MSORMs
-    r_msorms = all_camera_data.set_index(['timestamp', 'ROI']).unstack()['r_msorm'].reset_index()
+    r_msorms = (
```
Comment (EvanSimpson) [inline osmo_jupyter/calibration/do/prep.py:47]: I didn't notice this before, but I guess I prefer the `()` wrapper to
```
backslash() /
.continuation()
```

File: osmo_jupyter/plot/heatmap_test.py [hunk 2]
```diff
@@ -8,45 +8,44 @@
...
             [1, 1, 2, 2],
             [1, 1, 2, 2],
             [3, 3, 4, 4],
             [3, 3, 4, 4],
         ])
+        # fmt: on
 
         expected_block_centers = [(0.5, 0.5), (0.5, 2.5), (2.5, 0.5), (2.5, 2.5)]
+        # Disable autoformatting to maintain logical shape
+        # fmt: off
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py:46]: nit: you could just keep this as one larger `fmt: off` block

File: osmo_jupyter/plot/heatmap_test.py [hunk 3]
```diff
@@ -65,62 +64,47 @@ def test_cut_even_shape(self):
...
         np.testing.assert_array_equal(actual_block_centers, expected_block_centers)
         np.testing.assert_array_equal(actual_blocks, expected_blocks)
 
     def test_mismatch_shape_drops_edges(self):
-        mismatch_shaped_array = np.array([
-            [1, 1, 2],
-            [1, 1, 2],
-            [2, 2, 2],
-        ])
+        mismatch_shaped_array = np.array([[1, 1, 2], [1, 1, 2], [2, 2, 2]])
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py]: add `fmt: off` here?

File: osmo_jupyter/plot/heatmap_test.py [hunk 4]
```diff
@@ -65,62 +64,47 @@ def test_cut_even_shape(self):
...
+        )
 
         expected_block_centers = [(0.5, 0.5)]
-        expected_blocks = np.array([
-            [
-                [1, 1],
-                [1, 1]
-            ],
-        ])
+        expected_blocks = np.array([[[1, 1], [1, 1]]])
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py]: `fmt: off`?

File: osmo_jupyter/plot/heatmap_test.py [hunk 5]
```diff
@@ -65,62 +64,47 @@ def test_cut_even_shape(self):
...
-            [
-                [1, 2],
-                [3, 4]
-            ],
-            [
-                [5, 6],
-                [7, 8]
-            ],
-        ])
+        expected_blocks = np.array([[[1, 2], [3, 4]], [[5, 6], [7, 8]]])
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py]: `fmt: off`?

File: osmo_jupyter/plot/heatmap_test.py [hunk 6]
```diff
@@ -65,62 +64,47 @@ def test_cut_even_shape(self):
...
 
         np.testing.assert_array_equal(actual_block_centers, expected_block_centers)
         np.testing.assert_array_equal(actual_blocks, expected_blocks)
 
     def test_returns_nothing_if_block_size_larger_than_array(self):
-        small_array = np.array([
-            [1, 1, 1],
-            [1, 1, 1],
-        ])
+        small_array = np.array([[1, 1, 1], [1, 1, 1]])
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py]: `fmt: off`?

File: osmo_jupyter/plot/heatmap_test.py [hunk 7]
```diff
@@ -130,16 +114,7 @@ class TestGetBlockMeans2D:
...
-                [
-                    [1, 2],
-                    [3, 4],
-                ],
-                [
-                    [10, 20],
-                    [30, 40],
-                ]
-            ])
+            blocks=np.array([[[1, 2], [3, 4]], [[10, 20], [30, 40]]]),
```
Comment (jaimemarijke) [inline osmo_jupyter/plot/heatmap_test.py]: `fmt: off`?

**Review by jaimemarijke — APPROVED** [review]

File: docs/source/conf.py [hunk 8]
```diff
@@ -113,15 +112,12 @@
     # The paper size ('letterpaper' or 'a4paper').
     #
     # 'papersize': 'letterpaper',
-
```
Comment (rush340) [inline docs/source/conf.py:116]: It's making the formatting of these comments ugly by removing the blank lines. Maybe replace with blank `#` lines.

File: osmo_jupyter/plot/__init__.py [hunk 9]
```diff
@@ -1,7 +1,11 @@
 # Import from submodule to preserve backwards-compatibility with previous import paths.
 # e.g. `from plot import get_scatter`
 
-from .scatter import get_rgb_scatters, get_scatter, get_layout_with_annotations  # noqa: F401 unused imports
+from .scatter import (  # noqa: F401 unused imports
```
Comment (EvanSimpson) [inline osmo_jupyter/plot/__init__.py:4]: Does `noqa` still work if it's not on the same line as the unused import? Also, what is `noqa`?

Comment (jason-curtis) [reply osmo_jupyter/plot/__init__.py:4]: yes, this is the line it has to be on. It's a flake8 ignore.

File: osmo_jupyter/calibration/do/curve.py [hunk 10]
```diff
@@ -15,29 +18,27 @@ def _get_arrhenius_rate(temperature_c, preexponential_factor, activation_energy)
...
 
 def estimate_optical_reading_two_site_model_with_temperature(
-        do_and_temp,
-        f,
-        A_i0,
-        E_i0,
-        A_k_sv1,
-        A_k_sv2,
-        E_k_sv
+    do_and_temp, f, A_i0, E_i0, A_k_sv1, A_k_sv2, E_k_sv
```
Comment (jaimemarijke) [inline osmo_jupyter/calibration/do/curve.py]: `fmt: off` here?

File: osmo_jupyter/calibration/do/prep.py (hunk 1 above)
Comment (jason-curtis) [reply osmo_jupyter/calibration/do/prep.py:47]: yeah me too!

**Review by EvanSimpson — APPROVED** [review]

**Review by rush340 — APPROVED** [review]
Sad about the double quotes, but for the most part it's nice to have this automated.

**Pull request merged**